source/agora/node/Ledger.d

```diff
@@ -2101,21 +2134,13 @@ unittest
     auto mock_clock = new MockClock(params.GenesisTimestamp + 1);
     scope ledger = new TestLedger(WK.Keys.NODE2, blocks, params, 600.seconds, mock_clock);
 
-    Hash[] genesisEnrollKeys;
-    ledger.enroll_man.getEnrolledUTXOs(Height(1), genesisEnrollKeys);
-
-    // Reveal preimages for all validators but 1
-    foreach (idx, key; genesisEnrollKeys[0..$-1])
-    {
-        UTXO stake;
-        assert(ledger.utxo_set.peekUTXO(key, stake));
-        KeyPair kp = WK.Keys[stake.output.address];
-        auto cycle = PreImageCycle(kp.secret, params.ValidatorCycle);
-        const preimage = PreImageInfo(key, cycle[Height(params.ValidatorCycle)],
-                Height(params.ValidatorCycle));
+    // Add preimages for all validators (except for two of them) till end of cycle
+    auto skip_indexes = [ 2, 5 ];
 
-        ledger.enroll_man.addPreimage(preimage);
-    }
+    // Check that we are not testing skipping with node of test ledger as this will
+    // cause us problems as we assume the preimages are revealed for current ledger node
```

Comment:
I don't really understand this sentence ?

Reply:
Updated.

Reply:
``` D
scope ledger = new TestLedger(WK.Keys.NODE2, blocks, params, 600.seconds, mock_clock);

// Add preimages for all validators (except for two of them) till end of cycle
// Make sure not to pick enrollment index of NODE2 if used in TestLedger ctor above
auto skip_indexes = [ 2, 5 ];

// Make sure we are not simulating slashing for current node as we always assume the preimage is revealed for current ledger node
assert(!skip_indexes.canFind(ledger.enroll_man.getIndexOfEnrollment(Height(1))));
simulatePreimages(ledger.enroll_man, Height(params.ValidatorCycle), skip_indexes);

```